test/solver/pipe_cg_kernels.cpp

```diff
@@ -46,7 +46,8 @@ class PipeCg : public CommonTestFixture {
         // all vectors need the same stride as b, except x
         b = gen_mtx(size_m, size_n, size_n + 2);
         r = gen_mtx(size_m, size_n, size_n + 2);
-        z = gen_mtx(size_m, size_n, size_n + 2);
+        z1 = gen_mtx(size_m, size_n, size_n + 2);
+        z2 = gen_mtx(size_m, size_n, size_n + 2);
```

Comment:
```suggestion
        z1 = gen_mtx(size_m, size_n, size_n + 2);
        z2 = gko::clone(z1);
```
I think this is the reason. z1 and z2 is not equal from the beginning.
z1 and z2 should always be identical.

Reply:
after this, does the assignment still give wrong answer?

Reply:
no no, everything works well